src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,122 @@
+import Boom from 'boom';
+import { get } from 'lodash';
+
+import {
+  createFindQuery,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, body = {}) {
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return {
+      type: response._type,
+      id: response._id,
+      version: response._version,
+      attributes: body
+    };
+  }
+
+  async delete(type, id) {
+    const response = await this._withKibanaIndex('delete', {
+      type,
+      id,
+      refresh: 'wait_for'
+    });
+
+    if (get(response, 'found') === false) {
+      throw Boom.notFound();
+    }
+
+    return get(response, 'result') === 'deleted';
+  }
+
+  async find(options = {}) {
+    const {
+      search,
+      searchFields,
+      type,
+      fields,
+      perPage = 20,
+      page = 1,
+    } = options;
+
+    const esOptions = {
+      type,
+      _source: fields,
+      size: perPage,
+      from: perPage * (page - 1),
+      body: createFindQuery({ search, searchFields, type })
+    };
+
+    const response = await this._withKibanaIndex('search', esOptions);
+
+    return {
+      data: get(response, 'hits.hits', []).map(r => {
```

Comment:
Since we know that these are saved objects, we don't need to be quite so generic as `data` do we? `docs` or `objects` or `saved_objects` maybe?